jib-core/src/main/java/com/google/cloud/tools/jib/event/progress/DelayedConsumer.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.event.progress;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.BinaryOperator;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
```

Comment:
note to reviewers: this "delay" functionality (and the code) is factored out from `ListenableCountingOutputStream`.

Reply:
[Might be better called](https://blog.bitsrc.io/understanding-throttling-and-debouncing-973131c1ba07) "ThrottlingConsumer", but it's really a `ThrottlingDispatcher`? 

The _adder_ might be better called _reducer_, as in map-reduce.

Reply:
`ThrottlingConsumer` sounds good. As a general class for any `Consumer`, `Dispatcher` doesn't sound naturally to me.

About "reducer", I think it adversely adds confusion, since "reducer" usually means the "foldl" (or `Stream.reduce()`, `std::accumulate()`) that acts on a list while taking this binary operator, to "fold" or "reduce" them into a single element.